[Suggestions] Filter the suggestions to not show words with apostrophes

c0nst4ntin opened the pull request:
### What:

- [X] Bug Fix
- [ ] New Feature

### Description:

When coding we don't want to have suggestions like `checker's` but only `checkers`.
I did not add test, as I want to wait for some more pull requests to get merged, before I get a lot of conflicts.

**Before and After:**

<img width="873" alt="Bildschirmfoto 2025-01-06 um 23 16 13" src="https://github.com/user-attachments/assets/8ab78379-de53-4741-a7fa-d41d64e33d7e" />

### Related:

benjam-es reviewed (changes requested):
I believe the test cases need updating to match.

The concept of the PR makes a lot of sense to me, and the code looks good.

c0nst4ntin commented:
Thanks. Yes, however I will wait with the test otherwise there will be more conflicts. So I will add them later down the line when we are getting closer to this pull request

nunomaduro commented:
can you fix conflicts?

c0nst4ntin commented:
Fixed the conflicts, and also added a check for duplicate suggestions (as found in Issue #65).

-> I still didn't add the tests, as this would cause conflicts with #67
